### PR TITLE
Add a size getter to URLSearchParams

### LINKED ISSUE #163: size (or length) for URLSearchParams
`URLSearchParams` is basically a `Map` and should have a `size` property. The unfortunate alternative is:
```js
Array.from(url.searchParams).length
```

### PR DESCRIPTION
**Pull request by annevk**
Tests: https://github.com/web-platform-tests/wpt/pull/38655.

Fixes #163.

- [x] At least two implementers are interested (and none opposed):
   * Chromium
   * WebKit
- [x] [Tests](https://github.com/web-platform-tests/wpt) are written and can be reviewed and commented upon at:
   * https://github.com/web-platform-tests/wpt/pull/38655
- [x] [Implementation bugs](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) are filed:
   * Chromium: https://bugs.chromium.org/p/chromium/issues/detail?id=1418374
   * Gecko: https://bugzilla.mozilla.org/show_bug.cgi?id=1818269
   * WebKit: https://bugs.webkit.org/show_bug.cgi?id=252749
   * Deno: https://github.com/denoland/deno/issues/17882
   * Node.js: https://github.com/nodejs/node/pull/46308
- [x] [MDN issue](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) is filed: https://github.com/mdn/content/issues/24760

(See [WHATWG Working Mode: Changes](https://whatwg.org/working-mode#changes) for more details.)

***
<a href="https://whatpr.org/url/734.html" title="Last updated on Feb 22, 2023, 4:42 PM UTC (c02d79d)">Preview</a> | <a href="https://whatpr.org/url/734/fdaa0e5...c02d79d.html" title="Last updated on Feb 22, 2023, 4:42 PM UTC (c02d79d)">Diff</a>

### CODE REVIEW

**Comment by jasnell** [comment]
PRs opened for both Node.js and workerd (see links above)

**Comment by zloirock** [comment]
`core-js` PR https://github.com/zloirock/core-js/pull/1164

File: url.bs [hunk 1]
```diff
@@ -3484,6 +3486,9 @@ constructor steps are:</p>
  <li><p><a for=URLSearchParams>Initialize</a> <a>this</a> with <var>init</var>.
 </ol>
 
+<p>The <dfn attribute for=URLSearchParams><code>size</code></dfn> getter steps are to return
+<a>this</a>'s <a for=URLSearchParams>list</a>'s <a for=list>size</a>.
```
Comment (jasnell) [inline url.bs:3490]: I may have missed the nuance here. My interpretation of `size` is that it should return the total number of elements that would be returned by the iterator. Is that understanding correct? And if not, I think that's what it should be.

/cc @lucacasonato

Comment (annevk) [reply url.bs:3490]: Yeah that is the proposal as per the last set of comments in #163. Note that I haven't yet asked all browser implementers for feedback. I've been meaning to but some other things have higher priority.

**Comment by annevk** [comment]
@valenting @ricea is there interest from Gecko and Chromium in this API addition?

**Comment by ricea** [comment]
I need a compelling use case to put in the explainer.

**Comment by annevk** [comment]
One use case from the issue is checking whether there are any query parameters in a `URLSearchParams` object. You could get that answer through iteration or serialization, but a size check is much more natural.

**Comment by ricea** [comment]
@domenic Do you think you can find someone who's willing to push this through the Blink process?

**Comment by domenic** [comment]
If it's not you/your team, then I have no particular good ideas, sorry to say...

**Comment by ricea** [comment]
Okay. I will try to find a volunteer. In the meantime let's say Chrome is tentatively interested, to align with developer expectations for a map.

**Comment by bashi** [comment]
Does anyone have any plans to write WPTs? Chromium doesn't have strong interest but I can drive the Blink process if someone can write tests.

**Comment by annevk** [comment]
I wrote tests and filed implementation bugs. I asked for your review on the tests @bashi. If all looks in order I can merge it all.

**Comment by zloirock** [comment]
Added in [`core-js@3.29`](https://github.com/zloirock/core-js/releases/tag/v3.29.0).

**Pull request merged**